onnxscript/rewriter/pattern.py

```diff
@@ -999,6 +1006,11 @@ def _apply_to_graph_or_function(
         for rule in self.rules:
             deltas = []
             for i, node in enumerate(graph_or_function.nodes):
+                if (
+                    node.op_type not in self.rule_mapping
+                    or rule not in self.rule_mapping[node.op_type]
```

Comment:
This doesn't really give us much of a performance advantage. This is the unfortunate result of a concurrent PR change that swapped the order of the loops above (the earlier iteration was over nodes, and for every node iterate through matching rules).

We will need to swap the order of loops back to get the perf improvement. We will be able to do that by migrating to the new IR. The new IR will allow us to make a change to the graph, and continue iterating over the rest of the graph.

Reply:
I think you can switch the order of the loops. The second part of the loop is looking for conflicts. If two rules try to rewrite the same node, one will be applied. The other one will be tried again in the next iteration.

Reply:
cc @titaiwangms